index.js

```diff
@@ -16,7 +16,17 @@ module.exports = (iterable, opts) => {
 		nonull: true
 	}))
 		.map(x => path.resolve(x))
-		.filter(fs.lstatSync);
+		.filter(x => {
+			try {
+				return fs.lstatSync(x);
+			} catch (err) {
+				if (err.code === 'ENOENT') {
+					return false;
+				}
+
+				return Promise.reject(err);
```

Comment:
You can't return a rejected promise here. Instead, you should throw and rather wrap the module.export method in `Promise.resolve().then()` to catch thrown errors.

Reply:
Also, this predicate is kind of large, what about defining it outside of the filter call?

Reply:
> You can't return a rejected promise here.

Hm, that's right, although it does seem to work and reject on failures.

Reply:
@kevva Node.js logs the unhandled rejection, sure, but I wouldn't call that "working".